sdk/storage/Azure.Storage.Common/src/Shared/StorageResponseClassifier.cs

```diff
@@ -3,29 +3,42 @@
 
 using System;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Storage
 {
     internal class StorageResponseClassifier : ResponseClassifier
     {
-        public Uri SecondaryStorageUri { get; }
-
-        public StorageResponseClassifier(Uri secondaryStorageUri)
-        {
-            SecondaryStorageUri = secondaryStorageUri;
-        }
+        // The secondary URI to be used for retries on failed read requests
+        public Uri SecondaryStorageUri { get; set; }
 
         /// <summary>
-        /// Overridden version of IsRetriableResponse that allows for retrying 404 that occurs against the secondary host.
+        /// Overridden version of IsRetriableResponse that allows for Storage specific retry logic.
         /// </summary>
         /// <param name="message">The message containing both Response and Request</param>
         /// <returns></returns>
         public override bool IsRetriableResponse(HttpMessage message)
         {
-            if (message.Request.Uri.Host == SecondaryStorageUri.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
+            // If secondary storage Uri was specified, we want to retry if the current attempt was against the secondary Uri, and we
+            // get a response of NotFound. This is because the resource may not have been propagated to secondary Uri yet.
+            if (SecondaryStorageUri != null)
+            {
+                if (message.Request.Uri.Host == SecondaryStorageUri.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
```

Comment:
Nit - just `&&` the two `if`s together like

```c#
SecondaryStorageUri != null &&
message.Request.Uri.Host == SecondaryStorageUri.Host &&
message.Response.Status == Constants.HttpStatusCode.NotFound
```

Reply:
I thought this was a bit easier to read.